Dev (WIP) v0.0.5

erogol opened the pull request:
- [X] Initial Glow-TTS implementation
- [X] Universal FullBand-MelGAN vocoder.
- [X] New version Multi-Speaker model. (by @Edresson )
- [X] Improvements for Speaker Encoder. (by @mueller91)
- [x] Setting Tacotron losses weights from config.json.
- [x] Differential Spectral Loss 
- [x] French text cleaner. (by @WeberJulian)
- [x] Spanish model release. [Details](https://github.com/mozilla/TTS/issues/536).
- [x] French model release.

WeberJulian commented:
Hi @erogol,

I see that you plan to release a french model as well, here is the preprocess function that I use for french for mailabs. Paths structure differs depending on the language you use in mailabs.

```python
def mailabs(root_path, meta_files=None):
    """Normalizes M-AI-Labs meta data files to TTS format"""
    speaker_regex = re.compile("(male|female|mix)/(?P<speaker_name>[^/]+)/")
    if meta_files is None:
        csv_files = glob(root_path+"/**/metadata.csv", recursive=True)
    else:
        csv_files = meta_files
    # meta_files = [f.strip() for f in meta_files.split(",")]
    items = []
    for csv_file in csv_files:
        txt_file = os.path.join(root_path, csv_file)
        folder = os.path.dirname(txt_file)
        # determine speaker based on folder structure...
        speaker_name_match = speaker_regex.search(txt_file)
        if speaker_name_match is None:
            continue
        speaker_name = speaker_name_match.group("speaker_name")
        print(" | > {}".format(csv_file))
        with open(txt_file, 'r') as ttf:
            for line in ttf:
                cols = line.split('|')
                if meta_files is None:
                    wav_file = os.path.join(folder, 'wavs', cols[0] + '.wav')
                else:
                    wav_file = os.path.join(root_path,
                                            folder.replace("metadata.csv", ""),
                                            'wavs', cols[0] + '.wav')
                if os.path.isfile(wav_file):
                    text = cols[1].strip()
                    items.append([text, wav_file, speaker_name])
                else:
                    print("> File %s does not exist!" %(wav_file))
    return items
```
hope it helps

erogol commented:
> Hi @erogol,
> 
> I see that you plan to release a french model as well, here is the preprocess function that I use for french for mailabs. Paths structure differs depending on the language you use in mailabs.
> 
> ```python
> def mailabs(root_path, meta_files=None):
>     """Normalizes M-AI-Labs meta data files to TTS format"""
>     speaker_regex = re.compile("(male|female|mix)/(?P<speaker_name>[^/]+)/")
>     if meta_files is None:
>         csv_files = glob(root_path+"/**/metadata.csv", recursive=True)
>     else:
>         csv_files = meta_files
>     # meta_files = [f.strip() for f in meta_files.split(",")]
>     items = []
>     for csv_file in csv_files:
>         txt_file = os.path.join(root_path, csv_file)
>         folder = os.path.dirname(txt_file)
>         # determine speaker based on folder structure...
>         speaker_name_match = speaker_regex.search(txt_file)
>         if speaker_name_match is None:
>             continue
>         speaker_name = speaker_name_match.group("speaker_name")
>         print(" | > {}".format(csv_file))
>         with open(txt_file, 'r') as ttf:
>             for line in ttf:
>                 cols = line.split('|')
>                 if meta_files is None:
>                     wav_file = os.path.join(folder, 'wavs', cols[0] + '.wav')
>                 else:
>                     wav_file = os.path.join(root_path,
>                                             folder.replace("metadata.csv", ""),
>                                             'wavs', cols[0] + '.wav')
>                 if os.path.isfile(wav_file):
>                     text = cols[1].strip()
>                     items.append([text, wav_file, speaker_name])
>                 else:
>                     print("> File %s does not exist!" %(wav_file))
>     return items
> ```
> 
> hope it helps

Thx but I just choose a subset to obtain better voice consistency. And I just released the model. It'd be nice if you could check the model.
